Create new collection for Hcal lasermon digis + Hcal unpacker updates

jkunkle opened the pull request:
This pull request addresses three issues related to Hcal data unpacking.  (See more detailed explanations below) : 

1) Add a QIE10DigiCollection with tag "LASERMON" to keep laser monitoring data in the event
2) Improvements to the HcalUnpacker that are related to 1) 
3) Updates to RecoMET/METProducers/HcalNoiseInfoProducer to reflect the changes in 1)

Additional information to (1)
-------------------------------------
As we plan to use the lasermon channels to clean the data of any possible laser contamination, we desire to keep them in the emap so that we have a versioned accounting of them.  This requires changes to : 
- CalibFormats/HcalObjects/HcalTextToDetIdConverter, To parse the new emap entries from the .txt file
- DataFormats/HcalDetId/HcalCalibDetId, To add a unique CalibDetType
- EventFilter/HcalRawToDigi/HcalRawToDigi, to make the new collection and put it in the event
- EventFilter/HcalRawToDigi/HcalUnpacer, to unpack into the new collection

Additional information to (2)
-------------------------------------------
We often run into an issue with the HcalUnpacker where it encounters data having a different number of samples than the default collection (the number of samples is determined by the first FED of that type that is unpacked).  Currently the unpacker stops unpacking at this point and any data beyond that is not unpacked.  At this point in the code there are even comments that say "this is horrible".  The aim of this update is to avoid this situation and to provide access to this new data for private studies without making by-hand modifications to the unpacker.

This update does not change the default behavior of the unpacker except for 
1) The unpacker will no longer stop unpacking after encountering non-conforming data
2) the message that is printed in this case is modified to direct the user how to access the data

To access the data, the user can provide two lists to HcalRawToDigi, one that specifies the number of samples in the data, and a tag so the data can be stored in a unique collection.  There are two additional maps of integers (number of samples) to QIE10/11DataCollections.  The unpacker looks for these maps to be filled and unpacks accordingly.  Otherwise they follow the same processing as other QIE10/11 collections

Additional information to (3)
---------------------------------------
This just updates the code to read the lasermon channels from a QIE10DigiColleciton.  The algorithm does not change, there is just some structural reordering of the code to use the new collection

I have run the runTheMatrix tests and will soon provide instructions for testing the new features

Commit:
First attempt at HcalUnpacker improvements

Commit:
add Kevins suggestions

Commit:
bug fix

Commit:
partial update to include lasermon

Commit:
Lasermon unpacking is working

Commit:
update HcalNoiseInfoProducer

Commit:
fix

Commit:
remove extra if

cmsbuild commented:
The code-checks are being triggered in jenkins.

cmsbuild commented:
+code-checks

Logs: https://cmssdt.cern.ch/SDT/code-checks/cms-sw-PR-22385/3607

cmsbuild commented:
A new Pull Request was created by  @jkunkle  for master.

It involves the following packages:

CalibFormats/HcalObjects
DataFormats/HcalDetId
EventFilter/HcalRawToDigi
RecoMET/METProducers


@perrotta, @ghellwig, @civanch, @arunhep, @mdhildreth, @cmsbuild, @franzoni, @cerminar, @slava77, @lpernie can you please review it and eventually sign? Thanks.
@ghellwig, @jdamgov, @TaiSakuma, @gouskos, @ahinzmann, @mmarionncern, @rovere, @Martin-Grunewald, @jdolen, @nhanvtran, @gkasieczka, @tocheng, @schoef, @mmusich, @mariadalfonso, @seemasharmafnal this is something you requested to watch as well.
@davidlange6, @slava77, @fabiocos you are the release manager for this.

cms-bot commands are listed <a href="http://cms-sw.github.io/cms-bot-cmssw-cmds.html">here</a>

civanch commented:
please test

cmsbuild commented:
The tests are being triggered in jenkins.

Commit:
add protection for default qie1X collections

cmsbuild commented:
The code-checks are being triggered in jenkins.

jkunkle commented:
In discussing with colleagues and testing, Its clear that we needed to add a protection for the default qie10 and qie11 collections in the case that the data has a different number of samples.  This protection as assured in the past because the unpacker would return.  Here I just add the if statement that insures the protection

cmsbuild commented:
+code-checks

Logs: https://cmssdt.cern.ch/SDT/code-checks/cms-sw-PR-22385/3611

cmsbuild commented:
Pull request #22385 was updated. @perrotta, @ghellwig, @civanch, @arunhep, @mdhildreth, @cmsbuild, @franzoni, @cerminar, @slava77, @lpernie can you please check and sign again.

cmsbuild commented:
+1
Tested at: 27e3c78a445c14f018bf2d3ef53266480aa75c9b
https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-22385/26370/summary.html

cmsbuild commented:
Comparison job queued.

cmsbuild commented:
Comparison is ready
https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-22385/26370/summary.html

Comparison Summary:
 - No significant changes to the logs found
 - Reco comparison results: 4 differences found in the comparisons
 - DQMHistoTests: Total files compared: 29
 - DQMHistoTests: Total histograms compared: 2499373
 - DQMHistoTests: Total failures: 1
 - DQMHistoTests: Total nulls: 0
 - DQMHistoTests: Total successes: 2499196
 - DQMHistoTests: Total skipped: 176
 - DQMHistoTests: Total Missing objects: 0
 - DQMHistoSizes: Histogram memory added: 1.05000000004 KiB( 23 files compared)
 - Checked 118 log files, 9 edm output root files, 29 DQM output files

Commit:
Protect against QIE reset

cmsbuild commented:
The code-checks are being triggered in jenkins.

jkunkle commented:
In testing I found that we weren't properly handling the data when a QIE reset was present.  I've added a protection by checking the ok() bit of the data frame

cmsbuild commented:
+code-checks

Logs: https://cmssdt.cern.ch/SDT/code-checks/cms-sw-PR-22385/3624

cmsbuild commented:
Pull request #22385 was updated. @perrotta, @ghellwig, @civanch, @arunhep, @mdhildreth, @cmsbuild, @franzoni, @cerminar, @slava77, @lpernie can you please check and sign again.

perrotta commented:
please test

cmsbuild commented:
The tests are being triggered in jenkins.

cmsbuild commented:
Comparison job queued.

deguio commented:
hey @jkunkle 
since the plan is to test laser operations during cosmics data taking, could you please backport this PR to 10_0_x?
thanks,
F.

cmsbuild commented:
Comparison is ready
https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-22385/26398/summary.html

Comparison Summary:
 - No significant changes to the logs found
 - Reco comparison results: 0 differences found in the comparisons
 - DQMHistoTests: Total files compared: 29
 - DQMHistoTests: Total histograms compared: 2479021
 - DQMHistoTests: Total failures: 1
 - DQMHistoTests: Total nulls: 0
 - DQMHistoTests: Total successes: 2478844
 - DQMHistoTests: Total skipped: 176
 - DQMHistoTests: Total Missing objects: 0
 - DQMHistoSizes: Histogram memory added: 1.44999999995 KiB( 21 files compared)
 - Checked 118 log files, 9 edm output root files, 29 DQM output files

civanch commented:
+1

jkunkle commented:
Some instructions for testing 

1) Reproduce original problem with unpacker encountering data of different sample length.  
-----------------------------------------------------------------------
cmsrel CMSSW_10_1_X_2018-02-28-2300 #or a reasonably new release
cd CMSSW_10_1_X_2018-02-28-2300/src
cmsenv
cp -r /afs/cern.ch/user/j/jkunkle/public/HcalUnpackerCodeTest/* .
scram b
cmsRun TestUnpackerCode/TestUnpackerCode/test/test_unpacker_cfg.py  inputFiles=file:/afs/cern.ch/work/j/jkunkle/public/CMS/HCAL/HFLaser/5E197134-2D17-E811-95ED-02163E011C3B.root processEvents=10
# should see the message QIE10ZDC Collection has 3 samples per digi, raw data has 6!

2) Get the updated code
-----------------------------------------------------------------------
cd $CMSSW_BASE/src
rm -rf * #otherwise git will refuse to checkout packages
git cms-merge-topic -u 22385
cp -r /afs/cern.ch/user/j/jkunkle/public/HcalUnpackerCodeTest/* .
scram b -j8
cmsRun TestUnpackerCode/TestUnpackerCode/test/test_unpacker_cfg.py  inputFiles=file:/afs/cern.ch/work/j/jkunkle/public/CMS/HCAL/HFLaser/5E197134-2D17-E811-95ED-02163E011C3B.root processEvents=10 getLasermonDigis=1
# Since the lasermon channels are being handled we get no complaint from the unpacker.   The test code also prints out that it could access the 24 lasermon digis

3) Produce unpacker message.  Use a local run where there is an additional readout having a different number of time slices
-----------------------------------------------------------------------------
cmsRun TestUnpackerCode/TestUnpackerCode/test/test_unpacker_cfg.py  inputFiles=root://eoscms.cern.ch//eos/cms/store/group/dpg_hcal/comm_hcal/USC/run310190/USC_310190.root processEvents=10 getLasermonDigis=1 local=1 
# now one should see the message "The default QIE10 Collection has 6 samples per digi, while the current data has 10! " with additional instructions on how to store the data having 10 samples

4) Read out additional data
---------------------------------------------------------------------------
cmsRun TestUnpackerCode/TestUnpackerCode/test/test_unpacker_cfg.py  inputFiles=root://eoscms.cern.ch//eos/cms/store/group/dpg_hcal/comm_hcal/USC/run310190/USC_310190.root processEvents=10 getLasermonDigis=1 local=1 getUMNIODigis=1
# this adds the recommended configurations from the error message in 3) and reads out the 4 channels in the test script

5) The following items test the cases where one may misconfigure the code.  Test the case when a tag is requested that already exists (ie LASERMON already exists)
-------------------------------------------------------------------------
cmsRun TestUnpackerCode/TestUnpackerCode/test/test_unpacker_cfg.py  inputFiles=root://eoscms.cern.ch//eos/cms/store/group/dpg_hcal/comm_hcal/USC/run310190/USC_310190.root processEvents=10 getLasermonDigis=1 local=1 testTagDuplication=1
# should get the message "Cannot create an additional QIE10 sample with name LASERMON, it is already created!  It must be removed along with the corresponding entry in saveQIE10DataNSamples" In this case only the default QIE10 collections are created

6) Test the case when the configured nSamples duplicates an existing value
------------------------------------------------------------------------
cmsRun TestUnpackerCode/TestUnpackerCode/test/test_unpacker_cfg.py  inputFiles=root://eoscms.cern.ch//eos/cms/store/group/dpg_hcal/comm_hcal/USC/run310190/USC_310190.root processEvents=10 getLasermonDigis=1 local=1 testNSamplesDuplication=1
#should see the message "QIE10 data requested to be stored in tag UMNIO is already stored in the default QIE10 collection.  To avoid duplicating, remove the tag UMNIO from the saveQIE10DataTags and the value of 6 from the saveQIE10DataNSamples configurables to HcalRawToDigi"

7) Test when the vector inputs are mismatched in length
-----------------------------------------------------------------------
cmsRun TestUnpackerCode/TestUnpackerCode/test/test_unpacker_cfg.py  inputFiles=root://eoscms.cern.ch//eos/cms/store/group/dpg_hcal/comm_hcal/USC/run310190/USC_310190.root processEvents=10 getLasermonDigis=1 local=1 testMismatch=1
# should see the message "The saveQIE10DataTags and saveQIE10DataNSamples inputs must be the same length!  These will be ingored"

jkunkle commented:
For further testing, I just took a global run where we self trigger on the laser signal.   Data were recorded as in normal global runs, so we can check that the lasermon signal is propagated all the way to the MET filter.

I repeat the setup steps above (I added some additional code in my public area, so that should at least be copied again)
------------------------------------------------
cmsrel CMSSW_10_1_X_2018-02-28-2300 #or a reasonably new release
cd CMSSW_10_1_X_2018-02-28-2300/src
cmsenv
git cms-merge-topic -u 22385
cp -r /afs/cern.ch/user/j/jkunkle/public/HcalUnpackerCodeTest/* .
scram b -j8
cmsRun reco_RAW2DIGI_RECO.py 
cmsRun aod_PAT.py
cmsRun TestLaserCode/TestLaserCode/test/config_test.py 

There is a printout at the end of this code that checks that the METFilter bit corresponds correctly to the energy threshold that we want to apply.  It also creates histogram of the laser energy that can be found in lasermonHists.root

jkunkle commented:
Hi all,  One thing that bothers me somewhat is that this code will give error messages on every event in the default configuration (this was true in the past as well, but it was avoided because we didn't read out the hardware).  If this is a concern I can add a simple message counter which only prints the first N times (which I would put at 20).  How does this sound?

abdoulline commented:
Sounds reasonable (OK) to me...


On Fri, 2 Mar 2018, jkunkle wrote:

> 
> Hi all, One thing that bothers me somewhat is that this code will give error messages on every event in the default
> configuration (this was true in the past as well, but it was avoided because we didn't read out the hardware). If
> this is a concern I can add a simple message counter which only prints the first N times (which I would put at 20).
> How does this sound?
> 
> —
> You are receiving this because you are subscribed to this thread.
> Reply to this email directly, view it on GitHub, or mute the
> thread.[AEx02osHhZzFeMWkErKmsI2gft5LxqMlks5taS3NgaJpZM4SWSHf.gif]
> 
> 
>

Commit:
add message limit

cmsbuild commented:
The code-checks are being triggered in jenkins.

cmsbuild commented:
+code-checks

Logs: https://cmssdt.cern.ch/SDT/code-checks/cms-sw-PR-22385/3677

cmsbuild commented:
Pull request #22385 was updated. @perrotta, @ghellwig, @civanch, @arunhep, @mdhildreth, @cmsbuild, @franzoni, @cerminar, @slava77, @lpernie can you please check and sign again.

slava77 commented:
@cmsbuild please test

cmsbuild commented:
The tests are being triggered in jenkins.

cmsbuild commented:
+1
Tested at: 4c527fbc4d0cfb6a0b38a979a86ebcb783110d63
https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-22385/26474/summary.html

cmsbuild commented:
Comparison job queued.

cmsbuild commented:
Comparison is ready
https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-22385/26474/summary.html

Comparison Summary:
 - No significant changes to the logs found
 - Reco comparison results: 0 differences found in the comparisons
 - DQMHistoTests: Total files compared: 29
 - DQMHistoTests: Total histograms compared: 2479021
 - DQMHistoTests: Total failures: 1
 - DQMHistoTests: Total nulls: 0
 - DQMHistoTests: Total successes: 2478844
 - DQMHistoTests: Total skipped: 176
 - DQMHistoTests: Total Missing objects: 0
 - DQMHistoSizes: Histogram memory added: 1.03000000017 KiB( 21 files compared)
 - Checked 118 log files, 9 edm output root files, 29 DQM output files

jkunkle commented:
Hi All,

I have additionally tested by rerunning our online DQM to confirm that we can now have working monitoring (which was broken because the unpacker stops unpacking when it encounters data having differing number of samples).  I think that we in HCAL are happy with this update.  Can we aim to have it merged for the  10_1_0pre3 release?  Can I already make the pull request for backporting into 10_0_X?

perrotta commented on EventFilter/HcalRawToDigi/plugins/HcalRawToDigi.cc:
```diff
@@ -194,25 +272,72 @@ void HcalRawToDigi::produce(edm::Event& e, const edm::EventSetup& es)
+  for( const auto & addtl: colls.qie10Addtl ) {
+      if( addtl.second->samples() == colls.qie10->samples() ) {
+          std::string tag = saveQIE10Info_[addtl.second->samples()];
+          edm::LogError("HcalRawToDigi") << "QIE10 data requested to be stored in tag " 
```
Again: do these need to be LogError's?

perrotta commented on EventFilter/HcalRawToDigi/src/HcalUnpacker.cc:
```diff
@@ -903,3 +935,35 @@ void HcalUnpacker::unpackUMNio(const FEDRawData& raw, int slot, Collections& col
+  int limit = 50;
+  if( nPrinted_ >= limit ) {
+
+      if( nPrinted_ == limit ) edm::LogError("Invalid Data") << "Suppresing further error messages" << std::endl;
```
"Suppressing"...

perrotta commented on EventFilter/HcalRawToDigi/src/HcalUnpacker.cc:
```diff
@@ -903,3 +935,35 @@ void HcalUnpacker::unpackUMNio(const FEDRawData& raw, int slot, Collections& col
+
+  nPrinted_++;
+
+  int limit = 50;
```
Maybe the 20 that you suggested in the github thread can be enough...

perrotta commented on EventFilter/HcalRawToDigi/src/HcalUnpacker.cc:
```diff
@@ -903,3 +935,35 @@ void HcalUnpacker::unpackUMNio(const FEDRawData& raw, int slot, Collections& col
+              << conflict_ns << ") \nprocess.hcalDigis.save" << coll_type << "DataTags = cms.untracked.vstring( \"MYDATA\" )" ;
+  }
+
+  edm::LogError("Invalid Data") << message.str() << std::endl;
```
Again, is this needed to be a LogError?

perrotta commented:
@jkunkle : it is not from your update, but the static analyzer is reporting a dead initialization at L601 of EventFilter/HcalRawToDigi/src/HcalUnpacker.cc

In fact, the ```nps``` initialized there is never used, it being re-initialized later at L617

Could you (or someone else from the HCAL group) please check if this witnesses some bug in the code? If not, L601 (and the previous comment at L600) could simply be removed, just to avoid misunderstanding about what happens "for the 1.6 Gbps link data"

jkunkle commented:
Thanks, Andrea for the review.  I'll implement your suggestions (thanks for catching my spelling errors).  I've passed on the question about the issue with nps.  I'll wait until we have a resolution to that and push the changes together.

jkunkle commented:
@kpedro88  it looks like you made this second nps calculation in 2016, https://github.com/cms-sw/cmssw/commit/ec08635309798b575434550d117683d0d1810b1e.  Do you expect that we can remove the old calculation of nps?

kpedro88 commented:
yes, I think we can remove the old one.

Commit:
Change LogError to LogWarning, fix typos, remove unused nps calculation

cmsbuild commented:
The code-checks are being triggered in jenkins.

cmsbuild commented:
+code-checks

Logs: https://cmssdt.cern.ch/SDT/code-checks/cms-sw-PR-22385/3743

cmsbuild commented:
Pull request #22385 was updated. @perrotta, @ghellwig, @civanch, @arunhep, @mdhildreth, @cmsbuild, @franzoni, @cerminar, @slava77, @lpernie can you please check and sign again.

perrotta commented:
please test

cmsbuild commented:
+1
Tested at: f168a968f0d6c027bcb0d971b103aedf765427ca
https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-22385/26520/summary.html

cmsbuild commented:
Comparison job queued.

cmsbuild commented:
Comparison is ready
https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-22385/26520/summary.html

Comparison Summary:
 - No significant changes to the logs found
 - Reco comparison results: 2 differences found in the comparisons
 - DQMHistoTests: Total files compared: 29
 - DQMHistoTests: Total histograms compared: 2479021
 - DQMHistoTests: Total failures: 1
 - DQMHistoTests: Total nulls: 0
 - DQMHistoTests: Total successes: 2478844
 - DQMHistoTests: Total skipped: 176
 - DQMHistoTests: Total Missing objects: 0
 - DQMHistoSizes: Histogram memory added: 1.04000000007 KiB( 22 files compared)
 - Checked 118 log files, 9 edm output root files, 29 DQM output files

civanch commented:
+1

perrotta commented on RecoMET/METProducers/src/HcalNoiseInfoProducer.cc:
```diff
@@ -672,137 +649,180 @@ HcalNoiseInfoProducer::filldigis(edm::Event& iEvent, const edm::EventSetup& iSet
+      int ieta    = calibId.ieta();
+
+      // only check channels having the requested cboxch
+      if (std::find( laserMonCBoxList_.begin(), laserMonCBoxList_.end(),
```
Is this ```find``` needed? If ```cboxch``` is found, then the loop over the same collection is re-done below; if it is not found, the check is similar as the ```if( cboxch == laserMonCBoxList_[idx])``` which is right at the beginning of the ```for``` loop below

jkunkle replied:
yes, we discussed this the first time I committed this code and it didn't get removed.  I'll remove it.

perrotta commented on EventFilter/HcalRawToDigi/src/HcalUnpacker.cc:
```diff
@@ -404,7 +405,7 @@ void HcalUnpacker::unpackVME(const FEDRawData& raw, const HcalElectronicsMap& em
 	  valid=true;
 	}      
-	// add the word (if within settings or recent firmware [recent firmware ignores startSample/endSample])
+	// add the word (if within settings or recent firmware [recent firmware Suppressing startSample/endSample])
```
"Suppressing"? (maybe inadvertently replaced by a copy/paste?)

jkunkle replied:
must have been, I'll revert it

perrotta reviewed (commented):
With this I am done with the visual code check (I left the HcalNoiseInfoProducer behind last time).

Tomorrow, I will test this PR following your recipe

Commit:
remove additional check on cboxch, fix find and replace error

cmsbuild commented:
The code-checks are being triggered in jenkins.

cmsbuild commented:
+code-checks

Logs: https://cmssdt.cern.ch/SDT/code-checks/cms-sw-PR-22385/3829

cmsbuild commented:
Pull request #22385 was updated. @perrotta, @ghellwig, @civanch, @arunhep, @mdhildreth, @cmsbuild, @franzoni, @cerminar, @slava77, @lpernie can you please check and sign again.

perrotta commented:
please test

cmsbuild commented:
The tests are being triggered in jenkins.

jkunkle commented:
I no longer see my prescription for running tests in the thread here, is it just me?

abdoulline commented:
Me neither...
There is a box "67 hidden items, Load more..."
at position  between 8 days ago and 3 day ago, but 
clicking on it doesn't bring anything...


On Thu, 8 Mar 2018, jkunkle wrote:

> 
> I no longer see my prescription for running tests in the thread here, is it just me?
> 
> —
> You are receiving this because you are subscribed to this thread.
> Reply to this email directly, view it on GitHub, or mute the thread.[AEx02teXRGHDWyk6MfPKHw9CI5kusIKSks5tcPj4gaJpZM4SWSHf.gif]
> 
> 
>

perrotta commented:
The same happens to me: it looks like github does not dsplay any more all the messages in the thread hidden behind the button
![immagine](https://user-images.githubusercontent.com/4069749/37143549-c2d18e2e-22bb-11e8-8aa5-d4b2cdfe5210.jpg)

However, I can still access them through the direct link that I saved:
https://github.com/cms-sw/cmssw/pull/22385#issuecomment-369558236

jkunkle <notifications@github.com> ha scritto:

> I no longer see my prescription for running tests in the thread here, 
> is it just me?
>
> --
> You are receiving this because you were mentioned.
> Reply to this email directly or view it on GitHub:
> https://github.com/cms-sw/cmssw/pull/22385#issuecomment-371430075

jkunkle commented:
I ran the testing myself and found that the first step didn't work.  This is because I changed the emap, I've updated the testing code to read in the original emap for the first step

cmsbuild commented:
+1
Tested at: 9274e5396c8f303da8dd3c3125d3d6feca7dfc89
https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-22385/26688/summary.html

cmsbuild commented:
Comparison job queued.

cmsbuild commented:
Comparison is ready
https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-22385/26688/summary.html

Comparison Summary:
 - No significant changes to the logs found
 - Reco comparison results: 0 differences found in the comparisons
 - DQMHistoTests: Total files compared: 29
 - DQMHistoTests: Total histograms compared: 2480211
 - DQMHistoTests: Total failures: 1
 - DQMHistoTests: Total nulls: 0
 - DQMHistoTests: Total successes: 2480034
 - DQMHistoTests: Total skipped: 176
 - DQMHistoTests: Total Missing objects: 0
 - DQMHistoSizes: Histogram memory added: 1.10000000014 KiB( 22 files compared)
 - Checked 118 log files, 9 edm output root files, 29 DQM output files

perrotta commented:
+1
- Keep laser monitoring data in the event
- In case of data with extra samples in FED, those extra samples are now also unpacked and saved for further private studies (not being used for the current workflows)
- The chain of savings and warning messages designed for it had been tested with the test program provided (not in this PR)
- No change expected and observed in reco outputs in the standard workflows, while new output will get stored in case of lasermon events or extra samples in the FED

mariadalfonso commented:
@jkunkle 
now that is signed by RECO  to you want to make the backport in the 10_0 ?

We should have the backport in the 10_0_4 since we want to test these changes at P5 asap.

Thanks

fabiocos commented:
@lpernie @arunhep @ghellwig @civanch could you please check this PR?

civanch commented:
+1

lpernie commented:
+1

cmsbuild commented:
This pull request is fully signed and it will be integrated in one of the next master IBs (tests are also fine). This pull request will now be reviewed by the release team before it's merged. @davidlange6, @slava77, @smuzaffar, @fabiocos (and backports should be raised in the release meeting by the corresponding L2)

fabiocos commented:
+1
